addons/sale/models/account_move.py

```diff
@@ -64,6 +64,7 @@ def action_post(self):
 
         # We cannot change lines content on locked SO, changes on invoices are not forwarded to the SO if the SO is locked
         downpayment_lines = self.line_ids.sale_line_ids.filtered(lambda l: l.is_downpayment and not l.display_type and not l.order_id.locked)
+        downpayment_lines._compute_name()
```

Comment:
We also want to update the description of down payments if the order is locked.
```suggestion
        dp_lines = self.line_ids.sale_line_ids.filtered(lambda l: l.is_downpayment and not l.display_type)
        dp_lines._compute_name()  # Update the description of DP lines (Draft -> Posted)
        downpayment_lines = dp_lines.filtered(lambda sol: not sol.order_id.locked)
```

Reply:
Done! :100: